sql/slave.cc

```diff
@@ -3146,6 +3147,8 @@ void store_master_info(THD *thd, Master_info *mi, TABLE *table,
     if ((slave_timestamp= mi->rli.slave_timestamp))
     {
       (*field++)->store_timestamp((my_time_t) slave_timestamp, 0);
+      // primary's binlog timestamp should be >= replica's replication timestamp
+      DBUG_ASSERT(master_timestamp >= slave_timestamp);
```

Comment:
Huh, now that you've added this, `rpl_gtid_crash` fails at this assertion:

```
mariadbd: /home/buildbot/amd64-ubuntu-2004-debug/build/sql/slave.cc:3151: void store_master_info(THD*, Master_info*, TABLE*, String*): Assertion `master_timestamp >= slave_timestamp' failed.
250131  2:28:43 [ERROR] /home/buildbot/amd64-ubuntu-2004-debug/build/sql/mariadbd got signal 6 ;
```

I'd say we should defer the `Seconds_Behind_Master` changes to a future MDEV (though I'd say to file a JIRA so we can investigate the failure, i.e. is it a bug or is it somehow valid), as we've never had any complaints about SBM cutting off (and the new fields from MDEV-33856 are more accurate anyway).